Add Getting Started section (#449)
gardener/documentation: website/documentation/getting-started/observability/shoot-status.md

---
title: Shoot Status
---

## Overview

In this topic you can see various shoot statuses and how you can use them to monitor your workloads.

> ## Shoot Status 1

Reviewer: ```suggestion
## Shoot Status - Conditions
```

A shoot status contains conditions about the healthiness of your cluster.

You can retrieve it by using `kubectl get shoot -oyaml`

> ![](./images/shoot-status-1.png)

Reviewer: The same is true for the other pictures with status. If you need new screenshots please write me :)

> ## Shoot Status 2

Reviewer: ```suggestion
## Shoot Status - Constraints
```

The shoot status also contains constraints. If these constraints are met, your cluster operations are impaired and the cluster is likely to fail at some point. Please watch them and act accordingly.

![](./images/shoot-status-2.png)

> ## Shoot Status 3

Reviewer: ```suggestion
## Shoot Status - Last Operation
```

The `lastOperation`, `lastErrors`, and `lastMaintenance` give you information on what was last happening in your clusters. This is especially useful when you are facing an error.

In this example, nodes are being recreated and not all machines have reached the desired state.

![](./images/shoot-status-3.png)

## Shoot Status 4

> You can also see the status of the last credentials rotation. Here you can also programatically derive when you next need to trigger the rotation of credentials.

Reviewer: ```suggestion
You can also see the status of the last credentials rotation. Here you can also programatically derive when the last rotation was down in order to trigger the next rotation.
```

> In this example, nodes are being recreated and not all machines have reached the desired state.

Reviewer: ```suggestion
In this example, nodes are being recreated and not all machines have reached the desired state yet.
```
